lib/recurly/schema/json_deserializer.rb

```diff
@@ -27,13 +27,20 @@ def from_json(attributes = {})
             val = if val.is_a?(Hash) && !schema_attr.is_primitive?
                     schema_attr.recurly_class.from_json(val)
                   elsif val.is_a?(Array)
-                    val.map do |e|
+                    items = val.map do |e|
                       if e.is_a?(Hash) && !schema_attr.is_primitive?
                         schema_attr.recurly_class.from_json(e)
                       else
                         e
                       end
                     end
+                    # If the item type has a "currency" key, let's assume we can
+                    # safely index by currency and return a CurrencyArray
+                    if !schema_attr.is_primitive? && schema_attr.recurly_class.schema.get_attribute(:currency)
```

Comment:
For now, we are just looking to see if the item's type has a `currency` in the schema. In the future, I may make this a first class citizen in the recurly api spec.